src/Compilers/Core/Portable/Diagnostic/Diagnostic_SimpleDiagnostic.cs

```diff
@@ -130,11 +130,15 @@ public override bool Equals(object obj)
 
             public override int GetHashCode()
             {
-                return Hash.Combine(_descriptor,
-                        Hash.Combine(_messageArgs.GetHashCode(),
-                         Hash.Combine(_location.GetHashCode(),
-                          Hash.Combine(_severity.GetHashCode(), _warningLevel)
-                        )));
+                int hash = 0;
+
+                hash = Hash.Combine(_location.GetHashCode(), 0);
+                hash = Hash.Combine((int)_severity, hash);
+                hash = Hash.Combine(_warningLevel, hash);
+                hash = Hash.Combine(Hash.CombineValues(_messageArgs), hash);
+                hash = Hash.Combine(_descriptor, hash);
```

Comment:
Just an observation: While using the local variable undoubtedly makes the code easier to read, it does introduce a slight performance overhead. The original form's IL is shorter and (obviously) eliminates a local variable slot. The Roslyn compiler (currently) doesn't try to eliminate the local. I don't know whether the various JIT compilers will do so either.